indico/modules/attachments/controllers/event_package.py

```diff
@@ -102,14 +117,27 @@ def _generate_zip_file(self, attachments):
         temp_file = tempfile.NamedTemporaryFile('w')
         with ZipFile(temp_file.name, 'w', allowZip64=True) as zip_handler:
             for attachment in attachments:
-                if not attachment.folder.is_default:
-                    name = os.path.join(os.path.join(secure_filename(attachment.folder.title,
-                                                                     unicode(attachment.folder.id)),
-                                                     attachment.file.filename))
-                else:
-                    name = attachment.file.filename
-
+                name = self._prepare_folder_structure(attachment)
                 with attachment.file.storage.get_local_path(attachment.file.storage_file_id) as filepath:
                     zip_handler.write(filepath, name)
 
         return send_file('material-{}.zip'.format(self._conf.id), temp_file.name, 'application/zip', inline=False)
+
+    def _prepare_folder_structure(self, attachment):
+        linked_object = attachment.folder.linked_object
+        path_parts = [self._conf.getTitle()]
+
+        if isinstance(linked_object, SubContribution):
+            start_date = linked_object.parent.getAdjustedStartDate()
+            path_parts.extend([start_date.strftime('%Y%m%d_%A'),
+                               start_date.strftime('%H%M_{}').format(linked_object.parent.getTitle()),
```

Comment:
secure_filename(start_date.strftime('%H%M_{}').format(linked_object.parent.getTitle(), '')

no need for a good default here; with the prefix you'll always have something valid